gno.land/pkg/gnoweb/url.go

```diff
@@ -28,121 +32,147 @@ type GnoURL struct {
 	Query    url.Values // c=d
 }
 
-func (url GnoURL) EncodeArgs() string {
-	var urlstr strings.Builder
-	if url.Args != "" {
-		urlstr.WriteString(url.Args)
-	}
+// EncodeFlag is used to compose and encode URL components.
+type EncodeFlag int
 
-	if len(url.Query) > 0 {
-		urlstr.WriteString("?" + url.Query.Encode())
-	}
+const (
+	EncodePath EncodeFlag = 1 << iota
+	EncodeArgs
+	EncodeWebQuery
+	EncodeQuery
+	EncodeNoEscape // Disable escaping on arg
+)
 
-	return urlstr.String()
+// Has checks if the EncodeFlag contains all the specified flags.
+func (f EncodeFlag) Has(flags EncodeFlag) bool {
+	return f&flags != 0
 }
 
-func (url GnoURL) EncodePath() string {
+// Encode encodes the URL components based on the provided flags.
+// Encode assums the URL is valid.
+func (gnoURL GnoURL) Encode(encodeFlags EncodeFlag) string {
```

Comment:
I'm not convinced by using a flag based approach. What would you think of having more similar to netl/url.URL? (see the `String()` method)

Reply:
Hmm, that would not be possible. `url.URL` has a single representation defined by the structure, while GnoURL has different representations based on context and perspective. I used `flags` because it's handy, readable, composable, and helps define the logic in one place, as `url.URL` does for `String`. A compromise for me would be to unexpose `Encode(flags)` and only keep high-level methods like the one below the method.

I'm still open to replacing this, but I would like something similar to what I described in my comment in terms of features.